bokeh/driving.py

```diff
@@ -0,0 +1,146 @@
+''' Provide a set of decorators useful for repeatedly updating a
+a function parameter in a specified way each time the function is
+called.
+
+These decorators can be especially useful in conjunction with periodic
+callbacks in a Bokeh server application.
+
+Example:
+
+    As an example, consider the ``bounce`` forcing function, which
+    advances a sequence forwards and backwards:
+
+    .. code-block:: python
+
+        from bokeh.driving import bounce
+
+        @bounce([0, 1, 2])
+        def update(i):
+            print(i)
+
+    If this function is repeatedly called, it will print the following
+    sequence on standard out:
+
+    .. code-block:: none
+
+        0 1 2 2 1 1 0 0 1 2 2 1 ...
```

Comment:
Shouldn't this be:

    0 1 2 2 1 0 0 1 2 2 1 0

i.e. why does the sequence `1 1` ever appear?